File: caesar.py
import sys

def encrypt(n, text):
   """Encrypt text with Caesar cypher with rotation n."""

   es = ""

   for c in text:
      if (c >= "A") and (c <= "Z"):
         b = ord("A")
      elif (c >= "a") and (c <= "z"):
         b = ord("a")

      es += chr(((ord(c) - b + n) % 26) + b)

   return es

if __name__ == '__main__':
   if len(sys.argv) < 3:
      print "Usage: " + sys.argv[0] + " n text"
   else:
      print encrypt(int(sys.argv[1]), sys.argv[2])

File: caesar.rb
def encrypt(n, text)
   es = ""

   text.scan(/./) { |c|
      if c =~ /[A-Z]/ then
         b = "A"[0]
      elsif c =~ /[a-z]/ then
         b = "a"[0]
      end

      es << (((c[0] - b + n) % 26) + b)
   }

   return es
end

if ARGV.length < 2 then
   puts "Usage: ruby " + $0 + " n text"
else
   puts encrypt(ARGV[0].to_i(), ARGV[1])
end
